helper/util/restart_linux.go

```diff
@@ -69,8 +70,45 @@ func (*containerRestarter) RestartContainer() error {
 	}
 
 	// kill the process group
+	path := "/proc/" + strconv.Itoa(pgid)
+
+	stderrlog.Infof("Sending SIGINT...")
 	_ = syscall.Kill(-pgid, syscall.SIGINT)
-	time.Sleep(2000)
+	
+	for i := 0; i < 5; i++ {
+		time.Sleep(time.Second)
+		_, err := os.Stat(path)
+		if os.IsNotExist(err) {
+			stderrlog.Infof("Process terminated!")
+			return nil;
+		}
+	}
+
+	stderrlog.Infof("Sending SIGTERM...")
```

Comment:
You might consider moving the wait logic into a function. Something along the lines of:
```golang
stderrlog.Infof("Sending SIGINT...")
_ = syscall.Kill(-pgid, syscall.SIGINT)
if done, err := waitForTerminated(pgid); err != nil {
  return err
} else if done {
  return nil
}
```